config/config/src/main/java/io/helidon/config/BuilderImpl.java

```diff
@@ -335,6 +342,19 @@ private ConfigSource targetConfigSource(ConfigContext context) {
         return targetConfigSource;
     }
 
+    private boolean hasMapSource(String name) {
+        if (sources != null) {
+            String qualifiedName = "[" + name + "]";
+            for (ConfigSource source : sources) {
+                if (source instanceof MapConfigSource
+                    && source.description().contains(qualifiedName)) {
```

Comment:
If we rely on a description, you should also update the place the description is created to make sure it is not changed.
I would prefer a more deterministic way of finding these sources out - maybe wrapping them in a specific class (`EnvVarConfigSource` and `SysPropConfigSource` or similar) and looking for these types.

Reply:
Yes, agreed, this is a little fragile; I'm just about to lose internet for a couple of days, so will deal with this next week.